contracts/neutron_interchain_queries/src/contract.rs

```diff
@@ -113,3 +129,132 @@ pub fn sudo(deps: DepsMut<InterchainQueries>, env: Env, msg: SudoMsg) -> Contrac
+
+    deps.api.debug(
+        format!(
+            "WASMDEBUG: sudo_check_tx_query_result loaded query string; query_id: {:?},\
+             transactions_filter: {:?}",
+            query_id, transactions_filter,
+        )
+        .as_str(),
+    );
+
+    #[allow(clippy::match_single_binding)]
+    // Depending of the query type, check the transaction data to see whether is satisfies
+    // the original query. If you don't write specific checks for a transaction query type,
+    // all submitted results will be treated as valid.
+    //
+    // TODO: come up with solution to determine transactions filter type
+    match registered_query.registered_query.query_type.as_str() {
+        _ => {
+            // For transfer queries, query data looks like "{"transfer.recipient": "some_address"}"
+            let query_data: TransferRecipientQuery =
+                serde_json_wasm::from_str(transactions_filter.as_str())?;
+
+            deps.api.debug(
+                format!(
+                    "WASMDEBUG: sudo_check_tx_query_result parsed query string; query_id: {:?}",
+                    query_id
+                )
+                .as_str(),
+            );
+
+            let mut deposits: Vec<Transfer> = RECIPIENT_TXS
+                .load(deps.storage, query_data.recipient.as_str())
+                .unwrap_or_default();
+            for message in body.messages {
+                // Skip all messages in this transaction that are not Send messages.
+                if message.type_url != *COSMOS_SDK_TRANSFER_MSG_URL.to_string() {
+                    continue;
+                }
+
+                // Parse a Send message and check that it has the required recipient.
+                let transfer_msg: MsgSend = MsgSend::decode(message.value.as_slice())?;
+                if transfer_msg.to_address == query_data.recipient {
+                    deps.api.debug(
+                        format!(
+                            "WASMDEBUG: sudo_check_tx_query_result found a matching transaction; \
+                             query_id: {:?}, from_address: {:?}",
+                            query_id, transfer_msg.from_address,
+                        )
+                        .as_str(),
+                    );
+                    for coin in transfer_msg.amount {
+                        deposits.push(Transfer {
+                            sender: transfer_msg.from_address.clone(),
+                            amount: coin.amount,
+                            denom: coin.denom,
+                            recipient: query_data.recipient.clone(),
+                        });
+                    }
+                }
+            }
+
+            // If we didn't find a Send message with the correct recipient, return an error, and
+            // this query result will be rejected by Neutron: no data will be saved to state.
+            if deposits.is_empty() {
```

Comment:
But `deposits` won't be empty after the first transaction with transfer in it.

So after that even if we get transactions with no transfers, deposits will always contain transfer from previous txs because you read deposits from the storage, so this condition will be always false except when processing of the first tx.